### PR TITLE
Favorites AGL

### PR DESCRIPTION
**Pull request by alexlath**
* Adds Favorites model with init, count_of, and add_pet methods
* Adds FavoritesController#index, #update, #destroy actions.
* Adds favorites indicator link in navbar
* Adds way to add pet to favorites
* Adds way to delete all pets from favorites

100% test coverage

### CODE REVIEW

File: app/controllers/pets_controller.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ def index
 
   def show
     @pet = Pet.find(params[:id])
+    @favorites= Favorites.new(session[:favorites])
```
Comment (alexlath) [inline app/controllers/pets_controller.rb]: Will we need to add this to every action that has a corresponding view?

File: config/routes.rb [hunk 2]
```diff
@@ -36,4 +36,10 @@
   get '/shelters/:shelter_id/reviews/new', to: 'reviews#new'
   post '/shelters/:shelter_id', to: 'reviews#create'
   delete '/shelters/:shelter_id/reviews/:id', to: 'reviews#destroy'
+
+  # favorites
+  get '/favorites', to: 'favorites#index'
+
+  patch '/favorites/:pet_id', to: 'favorites#update'
+  delete '/favorites', to: 'favorites#destroy'
```
Comment (Maxwell-Baird) [inline config/routes.rb]: We should probably rename this to something that is clearer

File: app/controllers/pets_controller.rb (hunk 1 above)
Comment (Maxwell-Baird) [reply app/controllers/pets_controller.rb]: I think so

Comment (Maxwell-Baird) [reply app/controllers/pets_controller.rb]: I think so

**Pull request merged**